streaming/src/main/scala/org/apache/spark/streaming/dstream/DStreamCheckpointData.scala

```diff
@@ -87,7 +87,9 @@ class DStreamCheckpointData[T: ClassTag](dstream: DStream[T])
               if (fileSystem == null) {
                 fileSystem = path.getFileSystem(dstream.ssc.sparkContext.hadoopConfiguration)
               }
-              fileSystem.delete(path, true)
+              if (!fileSystem.delete(path, true)) {
+                logWarning(s"Error deleting old checkpoint file '$file' for time $time")
+              }
               timeToCheckpointFile -= time
               logInfo("Deleted checkpoint file '" + file + "' for time " + time)
```

Comment:
@gaborgsomogyi, wouldn't we avoid to log this when it's failed to remove?

Reply:
Wanted to make the smallest change but since it would touch another log statement its good idea.